File: ExampleScutiObjC/Pages/StoryboardViewController.m
//
//  StoryboardViewController.m
//  ExampleScutiObjC
//
//  Created by Adrian R on 26/09/2023.
//

#import "StoryboardViewController.h"
@import ScutiSDKSwift;

@interface StoryboardViewController () <ScutiSDKManagerDelegate>
@property (weak, nonatomic) IBOutlet UILabel *lblStoreReady;
@property (weak, nonatomic) IBOutlet UILabel *lblUserAuthenticated;
@property (weak, nonatomic) IBOutlet UILabel *lblCntProducts;
@property (weak, nonatomic) IBOutlet UILabel *lblCntRewards;

@end

@implementation StoryboardViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
}
- (void)viewWillAppear:(BOOL)animated {
    [super viewWillAppear:animated];
    ScutiSDKManager.shared.delegate = self;
    if (ScutiSDKManager.shared.scutiEventsObjC.isStoreReady) {
        self.lblStoreReady.text = @"Store Ready : ON";
        self.lblStoreReady.textColor = UIColor.greenColor;
    }
    if (ScutiSDKManager.shared.scutiEventsObjC.userToken != nil) {
        self.lblUserAuthenticated.text = @"User authenticated : YES";
        self.lblUserAuthenticated.textColor = UIColor.greenColor;
    }
    self.lblCntProducts.text = [NSString stringWithFormat:@"New Products : %ld", (long)ScutiSDKManager.shared.scutiEventsObjC.cntNewProducts];
    self.lblCntRewards.text = [NSString stringWithFormat:@"New Rewards : %ld", (long)ScutiSDKManager.shared.scutiEventsObjC.cntRewards];
}

/*
#pragma mark - Navigation

// In a storyboard-based application, you will often want to do a little preparation before navigation
- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
    // Get the new view controller using [segue destinationViewController].
    // Pass the selected object to the new view controller.
}
*/
#pragma mark - ScutiSDKManagerDelegate

- (void)onBackToGame {
    
}

- (void)onErrorOccurredWithError:(NSError * _Nonnull)error {
    
}

- (void)onLogout {
    self.lblUserAuthenticated.text = @"User authenticated : NO";
    self.lblUserAuthenticated.textColor = UIColor.redColor;
}

- (void)onNewProductsWithCntProducts:(NSInteger)cntProducts {
    self.lblCntProducts.text = [NSString stringWithFormat:@"New Products : %ld", (long)cntProducts];
}

- (void)onNewRewardsWithCntRewards:(NSInteger)cntRewards {
    self.lblCntRewards.text = [NSString stringWithFormat:@"New Rewards : %ld", (long)cntRewards];
}

- (void)onScutiButtonClicked {
    
}

- (void)onScutiExchangeWithExchange:(ScutiExchangeClass * _Nonnull)exchange {
    
}

- (void)onStoreReady {
    self.lblStoreReady.text = @"Store Ready : ON";
    self.lblStoreReady.textColor = UIColor.greenColor;
}

- (void)onUserTokenWithUserToken:(NSString * _Nonnull)userToken {
    self.lblUserAuthenticated.text = @"User authenticated : YES";
    self.lblUserAuthenticated.textColor = UIColor.greenColor;
}


@end

File: Scuti/Pages/StoryboardViewController.swift
//
//  StoryboardViewController.swift
//  ScutiSDKSwift
//
//  Created by mac on 21/09/2023.
//

import UIKit
import ScutiSDKSwift
import SwiftUI

//https://sarunw.com/posts/swiftui-view-as-uiview-in-storyboard/

class StoryboardViewController: UIViewController {
    @IBOutlet weak var lblStoreReady: UILabel!
    @IBOutlet weak var lblUserAuthenticated: UILabel!
    @IBOutlet weak var lblCntProducts: UILabel!
    @IBOutlet weak var lblCntRewards: UILabel!

    override func viewDidLoad() {
        super.viewDidLoad()

        // Do any additional setup after loading the view.
    }
    
    override func viewWillAppear(_ animated: Bool) {
        super.viewWillAppear(animated)
        ScutiSDKManager.shared.delegate = self
        if ScutiSDKManager.shared.scutiEvents.isStoreReady {
            lblStoreReady.text = "Store Ready : ON"
            lblStoreReady.textColor = .green
        }
        if ScutiSDKManager.shared.scutiEvents.userToken != nil {
            lblUserAuthenticated.text = "User authenticated : YES"
            lblUserAuthenticated.textColor = .green
        }
        lblCntProducts.text = "New Products : \(ScutiSDKManager.shared.scutiEvents.cntNewProducts)"
        lblCntRewards.text = "New Rewards : \(ScutiSDKManager.shared.scutiEvents.cntRewards)"
    }

    @IBSegueAction func embedScutiButton(_ coder: NSCoder) -> UIViewController? {
        let controller = UIHostingController(coder: coder, rootView: ScutiButton())
        controller?.view.backgroundColor = .clear
        return controller
    }

    /*
    // MARK: - Navigation

    // In a storyboard-based application, you will often want to do a little preparation before navigation
    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
        // Get the new view controller using segue.destination.
        // Pass the selected object to the new view controller.
    }
    */

}
extension StoryboardViewController: ScutiSDKManagerDelegate {
    func onStoreReady() {
        lblStoreReady.text = "Store Ready : ON"
        lblStoreReady.textColor = .green
    }
    func onLogout() {
        lblUserAuthenticated.text = "User authenticated : NO"
        lblUserAuthenticated.textColor = .red
    }
    func onNewProducts(cntProducts: Int) {
        lblCntProducts.text = "New Products : \(cntProducts)"
    }
    func onNewRewards(cntRewards: Int) {
        lblCntRewards.text = "New Rewards : \(cntRewards)"
    }
    func onUserToken(userToken: String) {
        lblUserAuthenticated.text = "User authenticated : YES"
        lblUserAuthenticated.textColor = .green
    }
}
